### ISSUE TITLE
[Bug] `mmdeploy.apis.pytorch2onnx.torch2onnx` with Call id: 0 failed. exit.

### ISSUE THREAD

**Issue by mmvxc** [open-mmlab/mmdeploy#2361]
### Checklist

- [X] I have searched related issues but cannot get the expected help.
- [X] 2. I have read the [FAQ documentation](https://github.com/open-mmlab/mmdeploy/tree/main/docs/en/faq.md) but cannot get the expected help.
- [X] 3. The bug has not been fixed in the latest version.

### Describe the bug

我参照[这里](https://github.com/open-mmlab/mmpose/tree/main/projects/rtmpose#tensorrt-1)，进行模型转换
mmengine - ERROR - C:\Users\pc\Desktop\thq\MMDeploy\mmdeploy\apis\core\pipeline_manager.py - pop_mp_output - 80 - `mmdeploy.apis.pytorch2onnx.torch2onnx` with Call id: 0 failed. exit.

### Reproduction

python tools/deploy.py \
    configs/mmdet/detection/detection_tensorrt_static-320x320.py \
    ../mmpose/projects/rtmpose/rtmdet/person/rtmdet_nano_320-8xb32_coco-person.py \
    https://download.openmmlab.com/mmpose/v1/projects/rtmpose/rtmdet_nano_8xb32-100e_coco-obj365-person-05d8511e.pth \
    demo/resources/human-pose.jpg \
    --work-dir rtmpose-trt/rtmdet-nano \
    --device cuda:0 \
    --show \
    --dump-info

### Environment

```Shell
08/18 14:36:21 - mmengine - INFO - **********Environmental information**********
08/18 14:36:24 - mmengine - INFO - sys.platform: win32
08/18 14:36:24 - mmengine - INFO - Python: 3.10.11 (tags/v3.10.11:7d4cc5a, Apr  5 2023, 00:38:17) [MSC v.1929 64 bit (AMD64)]
08/18 14:36:24 - mmengine - INFO - CUDA available: True
08/18 14:36:24 - mmengine - INFO - numpy_random_seed: 2147483648
08/18 14:36:24 - mmengine - INFO - GPU 0: NVIDIA GeForce RTX 2060
08/18 14:36:24 - mmengine - INFO - CUDA_HOME: C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v11.8
08/18 14:36:24 - mmengine - INFO - NVCC: Cuda compilation tools, release 11.8, V11.8.89
08/18 14:36:24 - mmengine - INFO - MSVC: 用于 x64 的 Microsoft (R) C/C++ 优化编译器 19.29.30151 版
08/18 14:36:24 - mmengine - INFO - GCC: n/a
08/18 14:36:24 - mmengine - INFO - PyTorch: 2.0.1+cu118
08/18 14:36:24 - mmengine - INFO - PyTorch compiling details: PyTorch built with:
  - C++ Version: 199711
  - MSVC 193431937
  - Intel(R) Math Kernel Library Version 2020.0.2 Product Build 20200624 for Intel(R) 64 architecture applications
  - Intel(R) MKL-DNN v2.7.3 (Git Hash 6dbeffbae1f23cbbeae17adb7b5b13f1f37c080e)
  - OpenMP 2019
  - LAPACK is enabled (usually provided by MKL)
  - CPU capability usage: AVX2
  - CUDA Runtime 11.8
  - NVCC architecture flags: -gencode;arch=compute_37,code=sm_37;-gencode;arch=compute_50,code=sm_50;-gencode;arch=compute_60,code=sm_60;-gencode;arch=compute_61,code=sm_61;-gencode;arch=compute_70,code=sm_70;-gencode;arch=compute_75,code=sm_75;-gencode;arch=compute_80,code=sm_80;-gencode;arch=compute_86,code=sm_86;-gencode;arch=compute_90,code=sm_90;-gencode;arch=compute_37,code=compute_37
  - CuDNN 8.7
  - Magma 2.5.4
  - Build settings: BLAS_INFO=mkl, BUILD_TYPE=Release, CUDA_VERSION=11.8, CUDNN_VERSION=8.7.0, CXX_COMPILER=C:/actions-runner/_work/pytorch/pytorch/builder/windows/tmp_bin/sccache-cl.exe, CXX_FLAGS=/DWIN32 /D_WINDOWS /GR /EHsc /w /bigobj /FS -DUSE_PTHREADPOOL -DNDEBUG -DUSE_KINETO -DLIBKINETO_NOCUPTI -DLIBKINETO_NOROCTRACER -DUSE_FBGEMM -DUSE_XNNPACK -DSYMBOLICATE_MOBILE_DEBUG_HANDLE, LAPACK_INFO=mkl, PERF_WITH_AVX=1, PERF_WITH_AVX2=1, PERF_WITH_AVX512=1, TORCH_DISABLE_GPU_ASSERTS=OFF, TORCH_VERSION=2.0.1, USE_CUDA=ON, USE_CUDNN=ON, USE_EXCEPTION_PTR=1, USE_GFLAGS=OFF, USE_GLOG=OFF, USE_MKL=ON, USE_MKLDNN=ON, USE_MPI=OFF, USE_NCCL=OFF, USE_NNPACK=OFF, USE_OPENMP=ON, USE_ROCM=OFF,

08/18 14:36:24 - mmengine - INFO - TorchVision: 0.15.2+cu118
08/18 14:36:24 - mmengine - INFO - OpenCV: 4.8.0
08/18 14:36:24 - mmengine - INFO - MMEngine: 0.8.3
08/18 14:36:24 - mmengine - INFO - MMCV: 2.0.1
08/18 14:36:24 - mmengine - INFO - MMCV Compiler: MSVC 192930148
08/18 14:36:24 - mmengine - INFO - MMCV CUDA Compiler: 11.8
08/18 14:36:24 - mmengine - INFO - MMDeploy: 1.2.0+a7e9be2
08/18 14:36:24 - mmengine - INFO -

08/18 14:36:24 - mmengine - INFO - **********Backend information**********
08/18 14:36:24 - mmengine - INFO - tensorrt:    8.5.1.7
08/18 14:36:24 - mmengine - INFO - tensorrt custom ops: Available
08/18 14:36:24 - mmengine - INFO - ONNXRuntime: None
08/18 14:36:24 - mmengine - INFO - pplnn:       None
08/18 14:36:24 - mmengine - INFO - ncnn:        None
08/18 14:36:24 - mmengine - INFO - snpe:        None
08/18 14:36:25 - mmengine - INFO - openvino:    None
08/18 14:36:25 - mmengine - INFO - torchscript: 2.0.1+cu118
08/18 14:36:25 - mmengine - INFO - torchscript custom ops:      NotAvailable
08/18 14:36:25 - mmengine - INFO - rknn-toolkit:        None
08/18 14:36:25 - mmengine - INFO - rknn-toolkit2:       None
08/18 14:36:25 - mmengine - INFO - ascend:      None
08/18 14:36:25 - mmengine - INFO - coreml:      None
08/18 14:36:25 - mmengine - INFO - tvm: None
08/18 14:36:25 - mmengine - INFO - vacc:        None
08/18 14:36:25 - mmengine - INFO -

08/18 14:36:25 - mmengine - INFO - **********Codebase information**********
08/18 14:36:25 - mmengine - INFO - mmdet:       3.1.0
08/18 14:36:25 - mmengine - INFO - mmseg:       None
08/18 14:36:25 - mmengine - INFO - mmpretrain:  None
08/18 14:36:25 - mmengine - INFO - mmocr:       None
08/18 14:36:25 - mmengine - INFO - mmagic:      None
08/18 14:36:25 - mmengine - INFO - mmdet3d:     None
08/18 14:36:25 - mmengine - INFO - mmpose:      None
08/18 14:36:25 - mmengine - INFO - mmrotate:    None
08/18 14:36:25 - mmengine - INFO - mmaction:    None
08/18 14:36:25 - mmengine - INFO - mmrazor:     None
08/18 14:36:25 - mmengine - INFO - mmyolo:      None
```

### Error traceback

```Shell
08/18 14:33:04 - mmengine - WARNING - Failed to search registry with scope "mmdet" in the "Codebases" registry tree. As a workaround, the current "Codebases" registry in "mmdeploy" is used to build instance. This may cause unexpected failure when running the built modules. Please check whether "mmdet" is a correct scope, or whether the registry is initialized.
08/18 14:33:04 - mmengine - WARNING - Failed to search registry with scope "mmdet" in the "mmdet_tasks" registry tree. As a workaround, the current "mmdet_tasks" registry in "mmdeploy" is used to build instance. This may cause unexpected failure when running the built modules. Please check whether "mmdet" is a correct scope, or whether the registry is initialized.
08/18 14:33:06 - mmengine - INFO - Start pipeline mmdeploy.apis.pytorch2onnx.torch2onnx in subprocess
08/18 14:33:07 - mmengine - WARNING - Failed to search registry with scope "mmdet" in the "Codebases" registry tree. As a workaround, the current "Codebases" registry in "mmdeploy" is used to build instance. This may cause unexpected failure when running the built modules. Please check whether "mmdet" is a correct scope, or whether the registry is initialized.
08/18 14:33:07 - mmengine - WARNING - Failed to search registry with scope "mmdet" in the "mmdet_tasks" registry tree. As a workaround, the current "mmdet_tasks" registry in "mmdeploy" is used to build instance. This may cause unexpected failure when running the built modules. Please check whether "mmdet" is a correct scope, or whether the registry is initialized.
Loads checkpoint by http backend from path: https://download.openmmlab.com/mmpose/v1/projects/rtmpose/rtmdet_nano_8xb32-100e_coco-obj365-person-05d8511e.pth
08/18 14:33:07 - mmengine - WARNING - DeprecationWarning: get_onnx_config will be deprecated in the future.
08/18 14:33:07 - mmengine - INFO - Export PyTorch model to ONNX: rtmpose-ort/rtmdet-nano\end2end.onnx.
C:\Users\pc\Desktop\thq\MMDeploy\mmdeploy\codebase\mmdet\models\detectors\single_stage.py:84: TracerWarning: Iterating over a tensor might cause the trace to be incorrect. Passing a tensor of different shape won't change the number of iterations executed (and might lead to errors or silently give incorrect results).
  img_shape = [int(val) for val in img_shape]
C:\Users\pc\Desktop\thq\MMDeploy\mmdeploy\codebase\mmdet\models\detectors\single_stage.py:84: TracerWarning: Converting a tensor to a Python integer might cause the trace to be incorrect. We can't record the data flow of Python values, so this value will be treated as a constant in the future. This means that the trace might not generalize to other inputs!
  img_shape = [int(val) for val in img_shape]
C:\Users\pc\Desktop\thq\MMDeploy\mmdeploy\core\optimizers\function_marker.py:160: TracerWarning: Converting a tensor to a Python integer might cause the trace to be incorrect. We can't record the data flow of Python values, so this value will be treated as a constant in the future. This means that the trace might not generalize to other inputs!
  ys_shape = tuple(int(s) for s in ys.shape)
C:\Users\pc\AppData\Local\Programs\Python\Python310\lib\site-packages\torch\functional.py:504: UserWarning: torch.meshgrid: in an upcoming release, it will be required to pass the indexing argument. (Triggered internally at ..\aten\src\ATen\native\TensorShape.cpp:3484.)
  return _VF.meshgrid(tensors, **kwargs)  # type: ignore[attr-defined]
C:\Users\pc\Desktop\thq\MMDeploy\mmdeploy\mmcv\ops\nms.py:416: TracerWarning: Converting a tensor to a Python boolean might cause the trace to be incorrect. We can't record the data flow of Python values, so this value will be treated as a constant in the future. This means that the trace might not generalize to other inputs!
  if not is_dynamic_batch(deploy_cfg) and batch_size == 1:
C:\Users\pc\Desktop\thq\MMDeploy\mmdeploy\mmcv\ops\nms.py:327: TracerWarning: torch.tensor results are registered as constants in the trace. You can safely ignore this warning if you use this function to create tensors out of constant variables that would be the same every time you call this function. In any other case, this might cause the trace to be incorrect.
  iou_threshold = torch.tensor([iou_threshold], dtype=torch.float32)
C:\Users\pc\Desktop\thq\MMDeploy\mmdeploy\mmcv\ops\nms.py:328: TracerWarning: torch.tensor results are registered as constants in the trace. You can safely ignore this warning if you use this function to create tensors out of constant variables that would be the same every time you call this function. In any other case, this might cause the trace to be incorrect.
  score_threshold = torch.tensor([score_threshold], dtype=torch.float32)
C:\Users\pc\Desktop\thq\MMDeploy\mmdeploy\pytorch\functions\topk.py:28: TracerWarning: torch.tensor results are registered as constants in the trace. You can safely ignore this warning if you use this function to create tensors out of constant variables that would be the same every time you call this function. In any other case, this might cause the trace to be incorrect.
  k = torch.tensor(k, device=input.device, dtype=torch.long)
C:\Users\pc\Desktop\thq\MMDeploy\mmdeploy\mmcv\ops\nms.py:44: TracerWarning: Converting a tensor to a Python float might cause the trace to be incorrect. We can't record the data flow of Python values, so this value will be treated as a constant in the future. This means that the trace might not generalize to other inputs!
  score_threshold = float(score_threshold)
C:\Users\pc\Desktop\thq\MMDeploy\mmdeploy\mmcv\ops\nms.py:45: TracerWarning: Converting a tensor to a Python float might cause the trace to be incorrect. We can't record the data flow of Python values, so this value will be treated as a constant in the future. This means that the trace might not generalize to other inputs!
  iou_threshold = float(iou_threshold)
C:\Users\pc\AppData\Local\Programs\Python\Python310\lib\site-packages\mmcv\ops\nms.py:123: TracerWarning: Converting a tensor to a Python boolean might cause the trace to be incorrect. We can't record the data flow of Python values, so this value will be treated as a constant in the future. This means that the trace might not generalize to other inputs!
  assert boxes.size(1) == 4
C:\Users\pc\AppData\Local\Programs\Python\Python310\lib\site-packages\mmcv\ops\nms.py:124: TracerWarning: Converting a tensor to a Python boolean might cause the trace to be incorrect. We can't record the data flow of Python values, so this value will be treated as a constant in the future. This means that the trace might not generalize to other inputs!
  assert boxes.size(0) == scores.size(0)
08/18 14:33:12 - mmengine - ERROR - C:\Users\pc\Desktop\thq\MMDeploy\mmdeploy\apis\core\pipeline_manager.py - pop_mp_output - 80 - `mmdeploy.apis.pytorch2onnx.torch2onnx` with Call id: 0 failed. exit.
```

**Comment by HichemMaiza** [comment]
@mmvxc could you verify that onnx and onnxruntime are installed in you environnement ? if not, could you try 
`pip install onnx onnxruntime`. tryto run again, I hope that this will help you solve the isse

**Comment by RunningLeon** [comment]
@mmvxc hi, could you check the shape of  boxes and scores in here?
> C:\Users\pc\AppData\Local\Programs\Python\Python310\lib\site-packages\mmcv\ops\nms.py:124: TracerWarning: Converting a tensor to a Python boolean might cause the trace to be incorrect. We can't record the data flow of Python values, so this value will be treated as a constant in the future. This means that the trace might not generalize to other inputs!
  assert boxes.size(0) == scores.size(0

**Comment by chemingmei** [comment]
请问这个问题解决了吗?

**Comment by xaclincoln** [comment]
Same issure here.  Executing rtmpose converting script on my windows 10 laptop (without CUDA environment).

**Error detail:**
> D:\python\lib\site-packages\mmengine\optim\optimizer\zero_optimizer.py:11: DeprecationWarning: `TorchScript` support for functional optimizers is deprecated and will be removed in a future PyTorch release. Consider using the `torch.compile` optimizer instead.
  from torch.distributed.optim import \
D:\python\lib\site-packages\mmengine\optim\optimizer\zero_optimizer.py:11: DeprecationWarning: `TorchScript` support for functional optimizers is deprecated and will be removed in a future PyTorch release. Consider using the `torch.compile` optimizer instead.
  from torch.distributed.optim import \
D:\python\lib\site-packages\albumentations\__init__.py:13: UserWarning: A new version of Albumentations is available: 2.0.5 (you have 1.4.18). Upgrade using: pip install -U albumentations. To disable automatic update checks, set the environment variable NO_ALBUMENTATIONS_UPDATE to 1.
  check_for_updates()
04/15 22:12:44 - mmengine - WARNING - Failed to search registry with scope "mmpose" in the "Codebases" registry tree. As a workaround, the current "Codebases" registry in "mmdeploy" is used to build instance. This may cause unexpected failure when running the built modules. Please check whether "mmpose" is a correct scope, or whether the registry is initialized.
04/15 22:12:44 - mmengine - WARNING - Failed to search registry with scope "mmpose" in the "mmpose_tasks" registry tree. As a workaround, the current "mmpose_tasks" registry in "mmdeploy" is used to build instance. This may cause unexpected failure when running the built modules. Please check whether "mmpose" is a correct scope, or whether the registry is initialized.
D:\python\lib\site-packages\mmengine\optim\optimizer\zero_optimizer.py:11: DeprecationWarning: `TorchScript` support for functional optimizers is deprecated and will be removed in a future PyTorch release. Consider using the `torch.compile` optimizer instead.
  from torch.distributed.optim import \
04/15 22:12:47 - mmengine - INFO - Start pipeline mmdeploy.apis.pytorch2onnx.torch2onnx in subprocess
D:\python\lib\site-packages\albumentations\__init__.py:13: UserWarning: A new version of Albumentations is available: 2.0.5 (you have 1.4.18). Upgrade using: pip install -U albumentations. To disable automatic update checks, set the environment variable NO_ALBUMENTATIONS_UPDATE to 1.
  check_for_updates()
04/15 22:12:51 - mmengine - WARNING - Failed to search registry with scope "mmpose" in the "Codebases" registry tree. As a workaround, the current "Codebases" registry in "mmdeploy" is used to build instance. This may cause unexpected failure when running the built modules. Please check whether "mmpose" is a correct scope, or whether the registry is initialized.
04/15 22:12:51 - mmengine - WARNING - Failed to search registry with scope "mmpose" in the "mmpose_tasks" registry tree. As a workaround, the current "mmpose_tasks" registry in "mmdeploy" is used to build instance. This may cause unexpected failure when running the built modules. Please check whether "mmpose" is a correct scope, or whether the registry is initialized.
Loads checkpoint by http backend from path: https://download.openmmlab.com/mmpose/v1/projects/rtmposev1/rtmpose-m_simcc-aic-coco_pt-aic-coco_420e-256x192-63eb25f7_20230126.pth
d:\目标追踪\mmpose\mmpose\datasets\datasets\utils.py:102: UserWarning: The metainfo config file "configs/_base_/datasets/coco.py" does not exist. A matched config file "d:\目标追踪\mmpose\mmpose\.mim\configs\_base_\datasets\coco.py" will be used instead.
  warnings.warn(
04/15 22:12:51 - mmengine - WARNING - DeprecationWarning: get_onnx_config will be deprecated in the future.
04/15 22:12:51 - mmengine - INFO - Export PyTorch model to ONNX: rtmpose-ort/rtmpose-m\end2end.onnx.
d:\目标追踪\mmdetection\mmdet\models\layers\se_layer.py:158: FutureWarning: `torch.cuda.amp.autocast(args...)` is deprecated. Please use `torch.amp.autocast('cuda', args...)` instead.
  with torch.cuda.amp.autocast(enabled=False):
d:\目标追踪\mmdetection\mmdet\models\backbones\csp_darknet.py:118: FutureWarning: `torch.cuda.amp.autocast(args...)` is deprecated. Please use `torch.amp.autocast('cuda', args...)` instead.
  with torch.cuda.amp.autocast(enabled=False):
04/15 22:12:56 - mmengine - ERROR - d:\目标追踪\mmdeploy\mmdeploy\apis\core\pipeline_manager.py - pop_mp_output - 80 - `mmdeploy.apis.pytorch2onnx.torch2onnx` with Call id: 0 failed. exit.
